services/core-web/src/components/noticeOfWork/applications/PermitPackage.js

```diff
@@ -0,0 +1,287 @@
+} from "@common/selectors/noticeOfWorkSelectors";
+import { EDIT_OUTLINE_VIOLET, EDIT_OUTLINE } from "@/constants/assets";
+import * as Permission from "@/constants/permissions";
+import { getDropdownNoticeOfWorkApplicationReviewTypeOptions } from "@common/selectors/staticContentSelectors";
+import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
+
+/**
+ * @constant ReviewNOWApplication renders edit/view for the NoW Application review step
+ */
+
+const propTypes = {
+  noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
+  importNowSubmissionDocumentsJob: PropTypes.objectOf(PropTypes.any),
+  updateNoticeOfWorkApplication: PropTypes.func.isRequired,
+  fetchImportedNoticeOfWorkApplication: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  openModal: PropTypes.func.isRequired,
+  setNoticeOfWorkApplicationDocumentDownloadState: PropTypes.func.isRequired,
+  isAdminView: PropTypes.bool,
+  isTableHeaderView: PropTypes.bool,
+};
+
+const defaultProps = {
+  isTableHeaderView: false,
+  isAdminView: false,
+  importNowSubmissionDocumentsJob: {},
+};
+
+export class PermitPackage extends Component {
+  state = {
+    cancelDownload: false,
+  };
+
+  createFinalDocumentPackage = (selectedCoreRows, selectedSubmissionRows) => {
+    const documentsPayload = this.props.noticeOfWork.documents.map((document) => {
+      document.is_final_package = selectedCoreRows.includes(
+        document.now_application_document_xref_guid
+      );
+      return document;
+    });
+
+    const submissionDocumentsPayload = this.props.noticeOfWork.filtered_submission_documents.map(
+      (document) => {
+        document.is_final_package = selectedSubmissionRows.includes(document.mine_document_guid);
+        return document;
+      }
+    );
+
+    const payload = {
+      ...this.props.noticeOfWork,
+      documents: documentsPayload,
+      submission_documents: submissionDocumentsPayload,
+    };
+
+    const message = "Successfully updated the final application package.";
+
+    this.props
+      .updateNoticeOfWorkApplication(payload, this.props.noticeOfWork.now_application_guid, message)
+      .then(() => {
+        this.props
+          .fetchImportedNoticeOfWorkApplication(this.props.noticeOfWork.now_application_guid)
+          .then(() => {
+            this.props.closeModal();
+          });
+      });
+  };
+
+  cancelDownload = () => {
+    this.setState({ cancelDownload: true });
+  };
+
+  downloadDocument = (url) => {
+    const a = document.createElement("a");
+    a.href = url.url;
+    a.download = url.filename;
+    a.style.display = "none";
+    document.body.append(a);
+    a.click();
+    a.remove();
+  };
+
+  waitFor = (conditionFunction) => {
+    const poll = (resolve) => {
+      if (conditionFunction()) resolve();
+      else setTimeout(() => poll(resolve), 400);
+    };
+
+    return new Promise(poll);
+  };
+
+  downloadDocumentPackage = () => {
+    const docURLS = [];
```

Comment:
Can you confirm that the download functionality and cancel download still works? I didn't test this feature.

Reply:
I have used it and as far as I can tell it works correctly.